### PR TITLE
Classes

### LINKED ISSUE #8: Write up Lesson Plan for Week 5 - Classes

### PR DESCRIPTION
**Pull request by georgeblahblah**
Adds slides for the Classes session.

Closes #8

### CODE REVIEW

File: 5-classes/slides.md [hunk 1]
```diff
@@ -0,0 +1,522 @@
...
+// ❌
+// Property 'a' is private in type 'Class1' but not in type 'Class2'
+const a: Class2 = new Class1();
+```
+
+---
+
+## Abstract
+
+The `abstract` modifier on fields and methods means they do not have an implementation provided. Abstract class members must exist on an `abstract` class.
```
Comment (mxdvl) [inline 5-classes/slides.md:507]: > Abstract class members must exist on an `abstract` class

What does that mean? Does it mean we need at least one? Or is it a typo and you meant to write that abstract members must have an implementation in the concrete class?

**Review by mxdvl — APPROVED** [review]

File: 5-classes/slides.md [hunk 2]
```diff
@@ -0,0 +1,522 @@
...
+---
+
+## Member Visibility
+
+Member visibility controls where we can see each member (fields and methods) from. There are 3 options:
+
+- `redadonly` fields. We cannot normally write to `readonly` fields
+- `public`. This is the default. We can see `public` members from anywhere
+- `protected`. We can see `protected` members from subclasses
+- `private`. We can only see `private` members from inside the declaring class
```
[the remark this answers is not in the document]
Comment (georgeblahblah) [reply 5-classes/slides.md:219]: yeah, and the generated JS is different too. TS generates a regular (public) property for `private`. It generates a hidden property using `WeakMap` for `#property`. I'll add a slide to disambiguate!

File: 5-classes/slides.md (hunk 1 above)
Comment (georgeblahblah) [reply 5-classes/slides.md:507]: oop! an example would help a lot here. It means this is not possible:

```ts
// ❌ This is a concrete class
class MyClass {
  abstract myProp: number = 0;
}

// ✅ This is an abstract class
abstract class MyClass {
  abstract myProp: number = 0 ;
}
```

**Review by JamieB-gu — APPROVED** [review]
🎉

File: 5-classes/slides.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (mxdvl) [reply 5-classes/slides.md:223]: I’ve just checked and this transformation does not happen if you [set the target to ES2022](https://www.typescriptlang.org/play?target=9#code/MYGwhgzhAECyCeBhcVoG8BQ1vQMQFt4AFAJwEsA3MAFwFNZb8AjWk6AXmgAYBuLAXwxA)

**Pull request merged**